emcc.py

```diff
@@ -2693,33 +2707,40 @@ def modularize():
   logger.debug('Modularizing, assigning to var ' + shared.Settings.EXPORT_NAME)
   src = open(final).read()
 
+  # TODO: exports object generation for MINIMAL_RUNTIME
+  exports_object = '{}' if shared.Settings.MINIMAL_RUNTIME else shared.Settings.EXPORT_NAME
+
   src = '''
 function(%(EXPORT_NAME)s) {
   %(EXPORT_NAME)s = %(EXPORT_NAME)s || {};
 
 %(src)s
 
-  return %(EXPORT_NAME)s;
+  return %(exports_object)s
 }
 ''' % {
     'EXPORT_NAME': shared.Settings.EXPORT_NAME,
-    'src': src
+    'src': src,
+    'exports_object': exports_object
   }
 
   if not shared.Settings.MODULARIZE_INSTANCE:
     # When MODULARIZE this JS may be executed later,
     # after document.currentScript is gone, so we save it.
     # (when MODULARIZE_INSTANCE, an instance is created
     # immediately anyhow, like in non-modularize mode)
-    src = '''
+    if shared.Settings.MINIMAL_RUNTIME and not shared.Settings.USE_PTHREADS:
+      src = 'var %s=%s' % (shared.Settings.EXPORT_NAME, src)
```

Comment:
why does this code path exist? should be commented at least.

Reply:
The other code path is redundant for singlethreaded applications. It creates an extra scope to capture `document.currentScript` immediately, when it is not needed or used anywhere in MINIMAL_RUNTIME programs. Added a comment.